### PR TITLE
[ADD] Wix Detox Types

### PR DESCRIPTION
**Pull request by TareqElMasriDev**
Types for [wix/detox](https://github.com/wix/detox)

### CODE REVIEW

**Comment by TareqElMasri** [comment]
@typescript-bot fixed 😄

File: types/detox/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "target": "ES2015",
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": true,
+        "strictFunctionTypes": false,
```
Comment (plantain-00) [inline types/detox/tsconfig.json]: It should be enabled for new packages.

File: types/detox/tslint.json [hunk 2]
```diff
@@ -0,0 +1,79 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
```
Comment (plantain-00) [inline types/detox/tslint.json]: This file should only `"extends": "dtslint/dt.json"` for new packages.

**Review by plantain-00 — CHANGES_REQUESTED** [review]

**Comment by mhegazy** [comment]
@TareqElMasri can you please address @plantain-00's comments

**Comment by mhegazy** [comment]
Thanks for your contribution. This PR has unaddressed comments and can not be merged in at the time being. For housekeeping purposes we are closing stale PRs. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

File: types/detox/tslint.json (hunk 2 above)
Comment (TareqElMasri) [reply types/detox/tslint.json]: Do you mean it shouldn't have any rules?

Comment (plantain-00) [reply types/detox/tslint.json]: Yes, just `{"extends": "dtslint/dt.json"}`.

**Comment by TareqElMasri** [comment]
@mhegazy @plantain-00 I have fixed the issues and pushed the changes in my fork, I will open a new PR.

**Pull request closed without merging**